server/mvcc/backend/tx_buffer.go

```diff
@@ -120,7 +120,7 @@ func newBucketBuffer() *bucketBuffer {
 func (bb *bucketBuffer) Range(key, endKey []byte, limit int64) (keys [][]byte, vals [][]byte) {
 	f := func(i int) bool { return bytes.Compare(bb.buf[i].key, key) >= 0 }
 	idx := sort.Search(bb.used, f)
-	if idx < 0 {
+	if idx < 0 || idx >= bb.used {
```

Comment:
Could you, please. add a test that exposes this situations,
 (or in practice for some reason it does not happen ?) 

How have you discovered this ?

Reply:
OK, I will try to add a test that exposes this situations later.